tests/tekton-resources/tasks/generators/clusterloader/load-pod-identity.yaml

```diff
@@ -175,6 +194,81 @@ spec:
+
+      REGION=$(params.region)
+      ENDPOINT_FLAG=""
+      if [ -n "$(params.endpoint)" ]; then
+        ENDPOINT_FLAG="--endpoint $(params.endpoint)"
+      fi
+
+      CLUSTER_NAME=$(params.cluster-name)
+      NAMESPACE=$(params.cl2-metric-namespace)
+      DIMENSION_NAME=$(params.cl2-metric-dimension-name)
+      DIMENSION_VALUE=$CLUSTER_NAME
+      METRIC_LATENCY_NAME=$(params.cl2-metric-latency-name)
+      PERIOD=$(params.cl2-metric-period)
+
+      # since the scalability test is running with the same cluster name, with cluster recreation
+      # it is important to know the range of start and end time to query metrics for the current run
+      # here we use cluster creation time start as start time and the current time as end time
+      START_TIME=$(aws eks $ENDPOINT_FLAG --region $REGION describe-cluster \
+        --name "$CLUSTER_NAME" \
+        --query "cluster.createdAt" \
+        --output text)
+
+      END_TIME=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+
+      response=$(aws cloudwatch get-metric-statistics \
+        --region "$REGION" \
+        --namespace "$NAMESPACE" \
+        --metric-name "$METRIC_LATENCY_NAME" \
+        --dimensions Name="$DIMENSION_NAME",Value="$DIMENSION_VALUE" \
+        --start-time "$START_TIME" \
+        --end-time "$END_TIME" \
+        --period "$PERIOD" \
+        --extended-statistics p50 p90 p99 \
+        --output json)
+
+      # extract p50 p90 p99 of credential fetching
+      latest=$(echo "$response" | jq -r '.Datapoints | sort_by(.Timestamp) | last')
+      p50=$(echo "$latest" | jq -r '."ExtendedStatistics"."p50" // "N/A"')
+      p90=$(echo "$latest" | jq -r '."ExtendedStatistics"."p90" // "N/A"')
+      p99=$(echo "$latest" | jq -r '."ExtendedStatistics"."p99" // "N/A"')
+
+      response=$(aws cloudwatch get-metric-statistics \
+        --region "$REGION" \
+        --namespace "$NAMESPACE" \
+        --metric-name "$METRIC_LATENCY_NAME" \
+        --dimensions Name="$DIMENSION_NAME",Value="$DIMENSION_VALUE" \
+        --start-time "$START_TIME" \
+        --end-time "$END_TIME" \
+        --period "$PERIOD" \
+        --statistics SampleCount \
+        --output json)
+
+      total_samples=$(echo "$response" | jq '[.Datapoints[].SampleCount] | add // 0')
+
+      # save metric results for s3 upload
+      cat <<EOF > eks_pod_identity_test_summary.json
+      {
+        "start_time": "$START_TIME",
+        "end_time": "$END_TIME",
+        "total_samples": $total_samples,
+        "p50": $p50,
+        "p90": $p90,
+        "p99": $p99
+      }
+      EOF
+
       # we expect to see all files from loadtest that clusterloader2 outputs here in this dir
       ls -larth
       aws s3 cp . s3://$S3_RESULT_PATH/  --recursive
+
+      # if p99 is equal to or more than 2 seconds, exit with failure
+      int_p99=$(echo "$p99" | awk '{printf "%d", $1}')
+      if [ "$int_p99" -lt 2 ]; then
+        echo "p99 is less than 2"
+      else
+        echo "p99 is 2 or more"
+        exit 1
```

Comment:
We don't want to exit and fail the test, we need to capture the test result like this  https://github.com/awslabs/kubernetes-iteration-toolkit/blob/3f5523db946e53dfcdabde926e03f8c06a434792/tests/tekton-resources/tasks/generators/clusterloader/load-slos.yaml#L171-L177


and use that to emit result like how we do it here - https://github.com/awslabs/kubernetes-iteration-toolkit/blob/3f5523db946e53dfcdabde926e03f8c06a434792/tests/tekton-resources/pipelines/eks/awscli-cl2-load-with-addons-slos.yaml#L360

So it can used configure alarm and cut tickets to your team.